src/functions-and-arrays.js

```diff
@@ -1,41 +1,147 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(a, b) {
+  if(a > b) {
+    return a;
+  } else {
+    return b;
+  }
+}
 
+maxOfTwoNumbers();
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(longestWord) {
+  if(longestWord.length === 0) {
+    return null;
+  } else {
+    let magicWord = "";
+    let longest = 0;
+    longestWord.forEach(function(element) {
+      if(longest < element.length) {
+        longest = element.length;
+        magicWord = element
+      }
+    });
+    return magicWord;
+  }
+}
 
+findLongestWord(words);
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(data) {
+  let total = 0;
+  for(let i = 0; i < data.length; i++) {
+    total = total + data[i];
+  }
+  return total;
+}
 
+sumNumbers(numbers);
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+const mixedArr2 = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function sum(mixArr) {
+  if(mixArr.length === 0) {
+    return 0
+  } else {
+    let numArr2 = [];
+    let sumNumArr = 0;
+    for(let w = 0; w < mixArr.length; w++) {
+      if(typeof mixArr[w] === 'boolean') {
+        if(mixArr[w]) {
+          numArr2.push(1)
+        } else {
+          numArr2.push(0)
+        }
+      } else if(typeof mixArr[w] === 'string') {
+        numArr2.push(mixArr[w].length)
+      } else {
+        numArr2.push(mixArr[w])
+      }
+    }
+    numArr2.forEach(function(valor) {
+      sumNumArr += valor;
+    })
+    return sumNumArr
```

Comment:
ya tenemos una funcion que hace la suma a un arreglo de numeros entonces te recomendaria no repetir y hacer uso de esta funcion 😉 
```suggestion
    return sumNumbers(numArr2);
```